networkx/algorithms/d_separation.py

```diff
@@ -140,3 +140,202 @@ def d_separated(G, x, y, z):
         return False
     else:
         return True
+
+
+@not_implemented_for("undirected")
+def minimal_d_separator(G, x, y):
+    """Compute a minimal d-separating set between X and Y.
```

Comment:
Can we use `u, v` instead of `x, y` to avoid confusion between nodes and sets of nodes?

Also, non of the functions (including the existing function) actually says what a d-separation is. It can't be more than a short paragraph, but it would sure be helpful. It could even go in the doc_string for the module if it is longer. Perhaps Wikipedia has some concise language.

And similarly, how hard would it be to have a one paragraph description of the algorithm. You only give the complexity here, and make the reader go to the paper for an inkling of what the algorithm does. Actually a paragraph describing what a "minimal d-separating set" might go a long way toward making this clear without details of the algorithm. At the moment, someone who hasn't heard this term before won't know what is going on.

Reply:
Added descriptions.